File: Client/client/src/app/_models/userParams.ts
import { User } from './user';

export class UserParams {
  currentUserName = '';
  gender= 0;
  minAge = 0;
  maxAge = 0;
  minHeight = 0;
  maxHeight = 0;
  province = 0;
  pageNumber = 1;
  minWeight = 0;
  maxWeight = 0;
  pageSize = 5;
  orderBy = 'lastActive';

  constructor(user?: User) {
    this.currentUserName = user?.userName || '';
  }
}

File: WebDating/Helpers/UserParams.cs
using WebDating.Entities.ProfileEntities;

namespace WebDating.Helpers
{
    public class UserParams : PaginationParams
    {
        public string CurrentUserName { get; set; }
        public Gender Gender { get; set; }
        public int MinAge { get; set; } = 18;
        public int MaxAge { get; set; } = 100;
        public int MinHeight { get; set; }
        public int MaxHeight { get; set; }
        public int MinWeight { get; set; }
        public int MaxWeight { get; set; }
        public string OrderBy { get; set; } = "lastActive";
        public Provice Province { get; set; }
    }
}
